src/segment/writer/mod.rs

```diff
@@ -248,8 +251,20 @@ impl Writer {
             // of the same key
             #[cfg(feature = "bloom")]
             if self.bloom_policy.is_active() {
-                self.bloom_hash_buffer
-                    .push(BloomFilter::get_hash(&item.key.user_key));
+                let key: Option<&[u8]> = match &self.opts.prefix_extractor {
```

Comment:
To control the bits per key independently for prefix and point reads, the prefix bloom filter needs to be built from its own buffer, and stored separately in the disk segment.
That's why there's a dedicated pointer for prefix filters in the segment metadata which is currently unused and always 0 (NULL): https://github.com/fjall-rs/lsm-tree/blob/77fb2a97f1707157f967f9dcbd047850aca3a5ca/src/segment/file_offsets.rs#L24